### DISCUSSION TITLE
Remote gets stuck Technic Hub after a few button presses

### DISCUSSION THREAD

**Discussion by mrinalraghupathi** [pybricks/support#1600] [Q&A]
I am putting pybricks code on my top gear car (set number 42109) to use it with the lego powered up remote control. The code below doesn't throw any exceptions. It runs for a while (about 400 - 500 iterations of the while loop) and works fine. Then the program just seems to get stuck in one iteration (almost as if there is a blocking call somwhere). The program at that point doesn't respond to any changes in button presses (even releasing all buttons isn't recognized). 

Not sure what I am doing wrong. Any pointers/help are much appreciated. Note: the remote works fine with pybricks on the cityhub on the app controlled batmobile (set 76112).

Firmware: Pybricks MicroPython v1.20.0-23-g6c633a8dd on 2024-03-21; LEGO Technic Hub with STM32L431RC
IDE version: v3.5.0b1 (Pybricks Beta v2.5.0-beta.2)

Code below

```python
from pybricks.hubs import TechnicHub
from pybricks.pupdevices import Motor, Remote
from pybricks.parameters import Button, Color, Direction, Port, Side, Stop
from pybricks.robotics import DriveBase
from pybricks.tools import wait, StopWatch

hub = TechnicHub()

motor_steer = Motor(Port.B) 
motor_drive = Motor(Port.D)
#hub.light.on(Color.RED)
motor_steer.run_angle(100, 0)
angle_right = motor_steer.run_until_stalled(200)
angle_left = motor_steer.run_until_stalled(-200)
# print(angle_right)
# print(angle_left)
center_angle = (angle_right + angle_left)/2
motor_steer.run_angle(200, (angle_right - angle_left)/2)
motor_steer.reset_angle(0)
#motor_steer.run_target(100, 0)
remote = Remote()
hub.light.on(Color.CYAN)
remote.light.on(Color.CYAN)

c = 0
while True:
    c += 1
    if divmod(c, 10)[1] == 0:
        # debugging to see if the loop is still running
        print(c)
    b = remote.buttons.pressed()
    if Button.LEFT_PLUS in b:
        print("go")
        motor_drive.run(1200)
    elif Button.LEFT_MINUS in b:
        print("back off")
        motor_drive.run(-1200)
    else:
        motor_drive.stop()

    if Button.RIGHT_PLUS in b:
        motor_steer.run_angle(200, 45)
    elif Button.RIGHT_MINUS in b:
        motor_steer.run_angle(200, -45) 
    else:
        motor_steer.run_angle(200, 0)
    wait(25)
```

**Comment by mrinalraghupathi** [comment]
As a follow up I changed 

```python 
motor_steer.run_angle(200, 45)
```

to 

```python
motor_steer.run_angle(200, 45, wait=False)
```

and this seems to have fixed the problem with blocking. However, the program now has the problem that releasing the buttons on the right side doesn't cause the steering to return to the center.

i.e. this line ```motor_steer.run_angle(200, 0)``` seems to have no effect

I am note sure why this is the case. The default `wait = True` might have blocked the rest of the program but I am not sure why this would happen given that the wheels did turn?

**Comment by mrinalraghupathi** [comment]
Realized the method I wanted is `run_target` not `run_angle`

**Reply by laurensvalk** [reply]
Glad it worked!

**Comment by laurensvalk** [comment]
Here's another program you could try:

```python
from pybricks.parameters import Button, Direction, Port
from pybricks.pupdevices import Motor, Remote
from pybricks.robotics import Car
from pybricks.tools import wait

# Set up all devices.

motor_steer = Motor(Port.B) 
motor_drive = Motor(Port.D)

car = Car(motor_steer, motor_drive)
rc = Remote(timeout=None)


# The main program starts here.
while True:
    # Control steering using the left - and + buttons
    car.steer(100 if Button.LEFT_PLUS in rc.buttons.pressed() else (-100 if Button.LEFT_MINUS in rc.buttons.pressed() else 0))
    # Control drive power using the right - and + buttons
    car.drive_power(100 if Button.RIGHT_PLUS in rc.buttons.pressed() else (-100 if Button.RIGHT_MINUS in rc.buttons.pressed() else 0))
    wait(50)


```

**Reply by mrinalraghupathi** [reply]
Thanks for this code, appreciate the help. I didn’t realize this was built-in. Although playing with the less abstracted code has been very rewarding and great for learning.
